pkg/apis/acme/v1/types_issuer.go

```diff
@@ -305,6 +313,19 @@ type ACMEChallengeSolverHTTP01IngressObjectMeta struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+type ACMEChallengeSolverHTTP01Istio struct {
+	// Namespace of the gateway that is used to generate the virtualservice for.
+	GatewayNamespace string `json:"gatewayNamespace"`
+
+	// Name of the gateway that is used to generate the virtualservice for.
+	GatewayName string `json:"gatewayName"`
+
+	// Optional pod template used to configure the ACME challenge solver pods
+	// used for HTTP01 challenges
+	// +optional
+	PodTemplate *ACMEChallengeSolverHTTP01IngressPodTemplate `json:"podTemplate,omitempty"`
```

Comment:
@wallrj moving this PodTemplate up in the yaml api tree would be cleaner. Unfortunately, I think this will make it harder to get this PR merged. This current solution is compatible with the old api. Ideally, we create a PR after getting this PR merged.

Reply:
Agreed. This can be improved in a later PR.

Reply:
See https://github.com/jetstack/cert-manager/issues/3942